substrate/frame/revive/src/benchmarking/code.rs

```diff
@@ -95,9 +95,28 @@ impl WasmModule {
 		Self::new(bench_fixtures::NOOP.to_vec())
 	}
 
-	/// A contract code that executes some ALU instructions in a loop.
-	pub fn instr() -> Self {
-		Self::new(bench_fixtures::INSTR.to_vec())
+	/// A contract code that does unaligned memory accessed in a loop.
+	pub fn instr(do_load: bool) -> Self {
+		let load = match do_load {
+			false => "",
+			true => "a0 = u64 [a0]",
+		};
+		let text = alloc::format!(
+			"
+		pub @deploy:
+		ret
+		pub @call:
+			@loop:
+				jump @done if t0 == a1
+				{load}
+				t0 = t0 + 1
+				jump @loop
+			@done:
+		ret
```

Comment:
Is `a0` the return register by the PVM calling convention? If not, it might be worthwhile to move its content into the return register (enforce a read), just to be absolutely sure nothing at any level might optimize the loads away.